src/pytorch_lightning/utilities/apply_func.py

```diff
@@ -342,7 +345,8 @@ def batch_to(data: Any) -> Any:
 
         kwargs = {}
         # Don't issue non-blocking transfers to CPU
-        if isinstance(data, Tensor) and device not in _CPU_DEVICES:
+        # Same with MPS due to a race condition bug: https://github.com/pytorch/pytorch/issues/83015
+        if isinstance(data, Tensor) and isinstance(device, torch.device) and device.type not in _BLOCKING_DEVICE_TYPES:
```

Comment:
```suggestion
        if isinstance(data, Tensor) and device.type not in _BLOCKING_DEVICE_TYPES:
```
`isinstance(device, torch.device)` is always true, given the new lines above that were added.

Reply:
but mypy does not understand this, so has to be kept probably

Reply:
No because `None` is also passed internally: https://github.com/Lightning-AI/lightning/pull/14368/#issuecomment-1225825455